tests/value/utils.py

```diff
@@ -23,3 +30,37 @@ def call_fn_multiple_seeds(
         A tuple of the results of the function.
     """
     return tuple(fn(*deepcopy(args), **deepcopy(kwargs), seed=seed) for seed in seeds)
+
+
+def measure_execution_time(
```

Comment:
Nit-picking: I would suggest renaming `measure_execution_time` (very descriptive name 👍🏽) as `timed` (less clear without context, but more concise when in use `timed(function)(args)`.

Less nit-picking: I would not expect a timer function to swallow all exceptions. If you want to report the time until an exception is raised, then raise it again after logging the time. Since you now cannot return a number, an alternative implementation could use function attributes instead of altering the return value. [Here's an example](https://gist.github.com/mdbenito/861c6593072cbdc511ef4a1a5a3da69e). Since this function might be useful for the performance benchmarks, we could move it somewhere in utils (don't create a new file just yet) and allow to keep track of the last n execution times in a history (configured with a decorator parameter).

Most importantly, only this last feature (which we don't need yet) would justify the existence of this function, because in python we already have `timeit` to do what `measure_execution_time` does, but with a lot of features it doesn't have (like setup code, averaging over n runs, disabling GC... 

I would therefore replace this by timeit for now.

Reply:
`timeit` doesn't deliver the result of the function. For our case this is not really beautiful. I like more the approach of the callable, because we could reuse that function also in the reproduction.

Reply:
😅Good point. What was I thinking